File: src/src/mockitoJava/Mocking/logic/ListClass.java
package mockitoJava.Mocking.logic;

import java.util.ArrayList;
import java.util.List;

public class ListClass {
    public ListService listService;

    public ListClass(ListService listService)
    {
        this.listService=listService;
    }

    public List<String> getListMethod(String match)
    {
        List<String> retrievedList1=listService.getList(match);
        List<String> retrievedList= new ArrayList<String>();
        for(String list : retrievedList1)
        {
            if(list.contains("Spring"))
            {
                retrievedList.add(list);
            }
        }
        return retrievedList;
    }
}

File: src/src/mockitoKotlin/logic/ListClass.kt
package mockitoKotlin.logic

import mockitoJava.Mocking.logic.ListService
import java.util.*


class ListClass(var listService: ListService) {
    fun getListMethod(match: String?): List<String> {
        val retrievedList1 = listService.getList(match)
        val retrievedList: MutableList<String> = ArrayList()
        for (list in retrievedList1) {
            if (list.contains("Spring")) {
                retrievedList.add(list)
            }
        }
        return retrievedList
    }
}
